### PR TITLE
Update clients

### PR DESCRIPTION
**Pull request by mmuckley**
This updates the client scripts for PyTorch Lightning 1.0.0 for the ongoing updates for Issue #84 -. It's another large PR, but hopefully makes the clients cleaner and more interpretable.

Summary of changes:
- Add help back to `argparse`.
- Create a new datamodule to independently handle data loader setup.
- Explicitly pass arguments rather than using `args`. This is better for linters.
- Path config is less forced on the user now - any arguments at the commend line for `data_path` and `log_path` should nicely override the defaults.
- Removed `configure_checkpoint` to make functionality more explicit in the client script.
- U-Net and VarNet transform classes moved to `fastmri.data.transforms`.
- Updates to requirements.
- Updates to tests.

### CODE REVIEW

**Review by michaldrozdzal — APPROVED** [review]
Matt, all looks good to me! I left you some comments that are mostly about clarifications and  adding comments in the UNet file to provide additional information for code users. Feel free to disregard my comments.

File: experimental/unet/train_unet_demo.py [hunk 1]
```diff
@@ -44,58 +77,85 @@ def build_args():
...
 
     # trainer config
     parser.set_defaults(
-        default_root_dir=default_root_dir,
-        checkpoint_callback=checkpoint_callback,
-        resume_from_checkpoint=resume_from_checkpoint,
         gpus=num_gpus,
         replace_sampler_ddp=False,
-        distributed_backend=backend,
+        accelerator=backend,
```
[the remark this answers is not in the document]
Comment (mmuckley) [reply experimental/unet/train_unet_demo.py:157]: This one actually comes from the PyTorch Lightning `Trainer`, so we don't have control over it...

https://pytorch-lightning.readthedocs.io/en/latest/trainer.html#accelerator

File: experimental/unet/train_unet_demo.py [hunk 2]
```diff
@@ -11,30 +11,63 @@
...
+        lr_step_size=args.lr_step_size,
+        lr_gamma=args.lr_gamma,
+        weight_decay=args.weight_decay,
+    )
 
-    # ------------------------
-    # 2 INIT TRAINER
-    # ------------------------
+    # ------------
+    # trainer
```
[the remark this answers is not in the document]
Comment (mmuckley) [reply experimental/unet/train_unet_demo.py:65]: Here we were trying to follow PyTorch Lightning code style for the client, but we can revert this if things aren't clear.

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/basic_examples/image_classifier.py

**Comment by mmuckley** [comment]
I've run a few integration tests at this point, training models and saving results. There could be some residual issues, but at this point I don't want to block the next few tasks so I'm going to merge. Thanks to @pjohnson519 for helping to review offline.

**Pull request merged**